community/CONTRIBUTING.md

```diff
@@ -5,6 +5,195 @@ Thanks for considering contributing to OPEA project. The contribution process is
 
 ## All The Ways To Contribute
 
+### Contribute a GenAIComponent
+
+1. Navigate to [OPEA GenAIComps](https://github.com/opea-project/GenAIComps) and locate the component folder your integration belongs to. If the microservice type already exists, review the [OPEA microservice API](https://opea-project.github.io/0.9.99/developer-guides/OPEA_API.html#opea-micro-service-api) and follow it in your implementation. Otherwise, if you are contributing a brand new microservice type, you need to define and contribute the API specifications. Please start by submit an RFC to get contributors feedback. 
+
+
+```
+GenAIComps
+├── comps
+│   ├── agent
+│   ├── asr
+│   ├── chathistory
+│   ├── cores
+│   │   ├── mega     #orchestrator, gateway, micro_service class code
+│   │   ├── proto    #api protocol
+│   │   └── telemetry
+│   ├── dataprep
+│   ├── embeddings
+│   ├── feedback_management
+│   ├── finetuning
+│   ├── guardrails
+│   ├── intent_detection
+│   ├── knowledgegraphs
+│   ├── llms
+│   ├── lvms
+│   ├── nginx
+│   ├── prompt_registry
+│   ├── ragas
+│   ├── reranks
+│   ├── retrievers
+│   ├── tts
+│   ├── vectorstores
+│   └── web_retrievers
+└── tests
+    ├── agent
+    ├── asr
+    ├── chathistory
+    ├── cores
+    ├── dataprep
+    ├── embeddings
+    ├── feedback_management
+    ├── finetuning
+    ├── guardrails
+    ├── intent_detection
+    ├── llms
+    ├── lvms
+    ├── nginx
+    ├── prompt_registry
+    ├── reranks
+    ├── retrievers
+    ├── test_finetuning_llm_pretraining.sh
+    ├── test_finetuning_rerank.sh
+    ├── test_reranks_mosec-neuralspeed.sh
+    ├── tts
+    ├── vectorstores
+    └── web_retrievers
+```
+
+2. Follow the folder structure in the TEI embedding component below:
+
+```
+GenAIComps
+├── comps
+│   └── embeddings
+│       ├── __init__.py
+│       └── tei     #vendor name or serving framework name
+│           ├── langchain
+│           │   ├── docker_compose_embedding.yaml  #could be multiple for different HW
+│           │   ├── Dockerfile        
+│           │   ├── Dockerfile_AMD_GPU         
+│           │   ├── Dockerfile_Nvidia_GPU      
+│           │   ├── embedding_tei.py    # definition and registration of microservice
+│           │   ├── README.md
+│           │   └── requirements.txt
+│           └── llama_index
+│               └── . . .
+├── tests
+│   └── embeddings
+│       ├── test_embeddings_tei_langchain.sh
+│       ├── test_embeddings_tei_langchain_amd_gpu.sh     
```

Comment:
test_embeddings_tei_langchain_on_amd_gpu.sh

Reply:
added _on_ With this is consistent with the tests in examples.